x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section/alert_details_app_section.tsx

```diff
@@ -59,7 +60,47 @@ export function AlertDetailsAppSection({
   rule,
   alert,
   timeZone,
+  setAlertSummaryFields,
 }: AlertDetailsAppSectionProps) {
+  useEffect(() => {
+    const alertSummaryFields = [
+      {
+        label: (
+          <FormattedMessage
+            id="xpack.apm.pages.alertDetails.alertSummary.actualValue"
+            defaultMessage="Actual value"
+          />
+        ),
+        value: formatAlertEvaluationValue(
+          alert?.fields[ALERT_RULE_TYPE_ID],
+          alert?.fields[ALERT_EVALUATION_VALUE]
+        ),
+      },
+      {
+        label: (
+          <FormattedMessage
+            id="xpack.apm.pages.alertDetails.alertSummary.expectedValue"
+            defaultMessage="Expected value"
+          />
+        ),
+        value: formatAlertEvaluationValue(
+          alert?.fields[ALERT_RULE_TYPE_ID],
+          alert?.fields[ALERT_EVALUATION_THRESHOLD]
+        ),
+      },
+      {
+        label: (
+          <FormattedMessage
+            id="xpack.apm.pages.alertDetails.alertSummary.source"
+            defaultMessage="Source"
+          />
+        ),
+        value: '-',
+      },
+    ];
+    setAlertSummaryFields(alertSummaryFields);
+  }, [alert?.fields, setAlertSummaryFields]);
+
   const params = rule.params;
   const environment = alert.fields[SERVICE_ENVIRONMENT];
   const latencyAggregationType = getAggsTypeFromRule(params.aggregationType);
```

Comment:
This component has a lot of different presentational widgets and holds a lot of logic for processing / fetching data for all these different widgets. In order to improve legibility I would strive to break it up into smaller pieces, and try to contain the logic that actually relates to a particular widget inside that separate component.

Reply:
💯 agree, and I would like to refactor it ASAP. I'm just wondering if this is right PR

Reply:
Done @CoenWarmer